File: frontend/src/types/counselor.ts
export interface CounselorStats {
  total_students: number;
  active_students: number;
  guest_sessions: number;
  open_escalations: number;
  acknowledged_escalations: number;
  resolved_escalations: number;
  total_assessments: number;
  high_severity_assessments: number;
  unread_notifications: number;
}

export interface EscalationTrendPoint {
  date: string;
  count: number;
}

export interface SeverityBreakdown {
  low: number;
  moderate: number;
  high: number;
  crisis: number;
}

export interface CounselorDashboardData {
  stats: CounselorStats;
  recent_escalations: Array<Record<string, unknown>>;
  escalation_trend: EscalationTrendPoint[];
  distress_breakdown: SeverityBreakdown;
  assessment_severity: Record<string, number>;
}

export interface StudentCaseloadItem {
  id: string;
  nickname: string | null;
  email: string | null;
  faculty: string | null;
  year_of_study: number | null;
  is_anonymous: boolean;
  escalation_count: number;
  open_escalations: number;
  latest_escalation_at: string | null;
  latest_assessment_severity: string | null;
}

export interface StudentAssessmentSummary {
  assessment_type: string;
  title: string;
  score: number;
  max_score: number;
  severity: string;
  completed_at: string;
}

export interface StudentDetail {
  student: {
    id: string;
    nickname: string | null;
    email: string | null;
    faculty: string | null;
    year_of_study: number | null;
    is_anonymous: boolean;
  };
  assessments: StudentAssessmentSummary[];
  escalations: Array<Record<string, unknown>>;
  mood_entries_count: number;
  checkins_count: number;
  chat_sessions_count: number;
}

File: backend/app/schemas/counselor.py
from pydantic import BaseModel

from app.schemas.escalation import EscalationStudentSummary


class CounselorStats(BaseModel):
    total_students: int
    active_students: int
    guest_sessions: int
    open_escalations: int
    acknowledged_escalations: int
    resolved_escalations: int
    total_assessments: int
    high_severity_assessments: int
    unread_notifications: int


class EscalationTrendPoint(BaseModel):
    date: str
    count: int


class SeverityBreakdown(BaseModel):
    low: int
    moderate: int
    high: int
    crisis: int


class CounselorDashboardResponse(BaseModel):
    stats: CounselorStats
    recent_escalations: list[dict]
    escalation_trend: list[EscalationTrendPoint]
    distress_breakdown: SeverityBreakdown
    assessment_severity: dict[str, int]


class StudentCaseloadItem(BaseModel):
    id: str
    nickname: str | None
    email: str | None
    faculty: str | None
    year_of_study: int | None
    is_anonymous: bool
    escalation_count: int
    open_escalations: int
    latest_escalation_at: str | None
    latest_assessment_severity: str | None


class StudentAssessmentSummary(BaseModel):
    assessment_type: str
    title: str
    score: int
    max_score: int
    severity: str
    completed_at: str


class StudentDetailResponse(BaseModel):
    student: EscalationStudentSummary
    assessments: list[StudentAssessmentSummary]
    escalations: list[dict]
    mood_entries_count: int
    checkins_count: int
    chat_sessions_count: int
